app/models/beard.rb

```diff
@@ -0,0 +1,35 @@
+class Beard < ActiveRecord::Base
+  belongs_to :user
+
+  STATUS = ['shaved', 'trimmed', 'unset']
+
+  validates :status,  presence: true
+  validates :user_id, presence: true
+
+  validate :valid_status
+
+  def update_status!(params)
+    self.status = params[:status] if params[:status]
+    self.save
+
+    self.toggle_massaged! if params[:is_massaged]
+  end
+
+  def toggle_massaged!
+    current_value = self.is_massaged
+    self.is_massaged = !!!current_value
+    self.save!
+  end
+
+  def massaged_label
+    is_massaged? ? 'Has massaged' : 'Has not massaged'
+  end
```

Comment:
I'd probably put it either in lib folder, create a different module that contains other 'Beard' functionalities. 

Also, this massage_label should've been on the helper, not in the model. 


> On Apr 29, 2015, at 4:41 AM, MoreRon <notifications@github.com> wrote:
> 
> In app/models/beard.rb:
> 
> > +  def update_status!(params)
> > +    self.status = params[:status] if params[:status]
> > +    self.save
> > +
> > +    self.toggle_massaged! if params[:is_massaged]
> > +  end
> > +
> > +  def toggle_massaged!
> > +    current_value = self.is_massaged
> > +    self.is_massaged = !!!current_value
> > +    self.save!
> > +  end
> > +
> > +  def massaged_label
> > +    is_massaged? ? 'Has massaged' : 'Has not massaged'
> > +  end
> If you would want to separate out view logic out of the data model how would you do it?
> 
> —
> Reply to this email directly or view it on GitHub.
>